### ISSUE TITLE
No longer working on soundcloud since yesterday (= June 26, 2019)

### ISSUE THREAD

**Issue by pitsi** [ytdl-org/youtube-dl#21550]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2019.06.27**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

I am not 100% sure about the last one, but I did search about issues on soundcloud and I did not find anything similar (= http error 500 on soundcloud).

## Verbose log
```
$ youtube-dl -v https://soundcloud.com/oneireuthslowbap/kantonamoiazei
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://soundcloud.com/oneireuthslowbap/kantonamoiazei']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.06.27
[debug] Python version 2.7.16 (CPython) - Linux-4.19.0-5-amd64-x86_64-with-debian-10.0
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, phantomjs 5
[debug] Proxy map: {}
[soundcloud] oneireuthslowbap/kantonamoiazei: Downloading webpage
WARNING: Unable to download webpage: HTTP Error 500: Internal Server Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/jim/bin/youtube-dl/__main__.py", line 19, in <module>
  File "/home/jim/bin/youtube-dl/youtube_dl/__init__.py", line 474, in main
  File "/home/jim/bin/youtube-dl/youtube_dl/__init__.py", line 464, in _real_main
  File "/home/jim/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 2008, in download
  File "/home/jim/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 796, in extract_info
  File "/home/jim/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
  File "/home/jim/bin/youtube-dl/youtube_dl/extractor/soundcloud.py", line 446, in _real_extract
UnboundLocalError: local variable 'info_json_url' referenced before assignment

```
## Description
I use debian testing x64 and I was on youtube-dl 2019.06.21 when I noticed the issue yesterday afternoon. It was working as it should in the morning, so I assume soundcloud changed something in its site and broke youtube-dl.
I downloaded the newer version today and tested it on the same url, but it still fails to download. The previous one was also tested on 2 different arch installations and the result was the same.
 
I mainly use youtube-dl with mpv, like so
`mpv --shuffle https://soundcloud.com/oneireuthslowbap`
Mpv uses youtube-dl to scrape the site and play the content.

Thank you in advance.

**Comment by pitsi** [comment]
Can you please write the reason for closing my bug report or what is it duplicate to?

**Comment by pitsi** [comment]
If I understand correctly, with commit c5606802474822887b75af7de23de6679264c0fa, soundcloud functionality should work now.

I manually patched line 224 of youtube_dl/extractor/soundcloud.py with the new client id string. Proof
```
$ cat /usr/lib/python3/dist-packages/youtube_dl/extractor/soundcloud.py | grep -n "_CLIENT_ID"
224:    _CLIENT_ID = 'BeGVhOrGmfboy1LtiHTQF6Ejpt9ULJCI'
234:        return 'https://api.soundcloud.com/resolve.json?url=' + url + '&client_id=' + cls._CLIENT_ID
273:        query = {'client_id': self._CLIENT_ID}
402:            info_json_url = 'https://api.soundcloud.com/tracks/' + track_id + '.json?client_id=' + self._CLIENT_ID
518:            'client_id': self._CLIENT_ID,
719:            'client_id': self._CLIENT_ID,
763:        query['client_id'] = self._CLIENT_ID
```
But youtube-dl still fails, with the same error
```
$ youtube-dl -v https://soundcloud.com/oneireuthslowbap/kantonamoiazei
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://soundcloud.com/oneireuthslowbap/kantonamoiazei']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2019.06.27
[debug] Python version 3.7.3 (CPython) - Linux-4.19.0-5-amd64-x86_64-with-debian-10.0
[debug] exe versions: ffmpeg 4.1.3, ffprobe 4.1.3, phantomjs 5
[debug] Proxy map: {}
[soundcloud] oneireuthslowbap/kantonamoiazei: Downloading webpage
WARNING: Unable to download webpage: HTTP Error 500: Internal Server Error
Traceback (most recent call last):
  File "/usr/bin/youtube-dl", line 11, in <module>
    load_entry_point('youtube-dl==2019.6.27', 'console_scripts', 'youtube-dl')()
  File "/usr/lib/python3/dist-packages/youtube_dl/__init__.py", line 479, in main
    _real_main(argv)
  File "/usr/lib/python3/dist-packages/youtube_dl/__init__.py", line 469, in _real_main
    retcode = ydl.download(all_urls)
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 2008, in download
    url, force_generic_extractor=self.params.get('force_generic_extractor', False))
  File "/usr/lib/python3/dist-packages/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/youtube_dl/extractor/soundcloud.py", line 446, in _real_extract
    info_json_url, full_title, 'Downloading info JSON')
UnboundLocalError: local variable 'info_json_url' referenced before assignment

```
So please check it again. I would post on issue #21546, but it is closed for further comments.

p.s. As seen on the output of youtube-dl -v, the other issue report uses some prehistoric version of youtube-dl from 2017 (2017.09.24), thus I think mine is better, because I did not lie when I ticked the "I've verified that I'm running youtube-dl version 2019.06.27" checkbox.

**Comment by bitraid** [comment]
If you check the original issue, you'll find out that the fix is f7a147e.
